src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIResponsesChatClient.cs

```diff
@@ -226,6 +230,34 @@ internal static IEnumerable<ChatMessage> ToChatMessages(IEnumerable<ResponseItem
         }
     }
 
+    private static DataContent GetContentFromImageGen(ImageGenerationCallResponseItem outputItem, ResponseCreationOptions? options)
+    {
+        var imageGenTool = options?.Tools.OfType<ImageGenerationTool>().FirstOrDefault();
+        string outputFormat = imageGenTool?.OutputFileFormat?.ToString() ?? "png";
+
+        return new DataContent(outputItem.GeneratedImageBytes, $"image/{outputFormat}")
+        {
+            RawRepresentation = outputItem
+        };
+    }
+
+    private static DataContent GetContentFromImageGenPartialImageEvent(StreamingResponseImageGenerationCallPartialImageUpdate update, ResponseCreationOptions? options)
+    {
+        var imageGenTool = options?.Tools.OfType<ImageGenerationTool>().FirstOrDefault();
+        var outputType = imageGenTool?.OutputFileFormat?.ToString() ?? "png";
+
+        return new DataContent(update.PartialImageBytes, $"image/{outputType}")
+        {
+            RawRepresentation = update,
+            AdditionalProperties = new()
+            {
+                [nameof(update.ItemId)] = update.ItemId,
+                [nameof(update.OutputIndex)] = update.OutputIndex,
+                [nameof(update.PartialImageIndex)] = update.PartialImageIndex
+            }
+        };
+    }
```

Comment:
We've begun adopting a pattern where the server tools have call and result types, e.g.
McpServerToolCallContent and McpServerToolResultContent
CodeInterpreterToolCallContent and CodeInterpreterToolResultContent

We'll want to do the same thing for HostedWebServerTool, so WebServerToolCallContent and WebServerToolResultContent.

We should do the same for image generation as well. That would mean adding an ImageGenerationToolCallContent and ImageGenerationToolResultContent, and these generated images would be available in a property hanging off of the ImageGenerationToolResultContent.

Reply:
Got it.  I will fix that up.  I think that makes the ImageGeneratingChatClient a bit easier to implement too.